### ISSUE TITLE
Replace ember-validations

### ISSUE THREAD

**Issue by internets** [HospitalRun/hospitalrun-frontend#1382]
```ember-validations``` is no longer under active development, and has become a blocker for completion of #1370 (ember-getowner-polyfill.import deprecation). There's a PR that's over a year old that is unlikely to ever be accepted: DockYard/ember-validations#442. 

The way I see it there are two options: fork or replace.

If fork, it's possible that forking and cherry-picking the commit from that PR will solve the immediate problem.

If replace, the two most common actively developed validation solutions to my knowledge are ember-cp-validations and ember-changeset-validations. I recommend ember-cp-validations for the following reasons:

- The broad strokes of the API are very similar to ember-validations
- It doesn't require buying into the ember-changeset way of handling form objects

**Comment by broerse** [comment]
@martndemus can you comment on what state `ember-validations` is in. Is it on the DockYard roadmap to update it to Ember 3.0?

**Comment by martndemus** [comment]
We are not planning to do anything about ember-validations.

**Comment by stukalin** [comment]
Well, it seems to be a large piece of work, migrating it. I've started the new branch but the process is not so straightforward. See https://github.com/HospitalRun/hospitalrun-frontend/compare/ember-validations?expand=1

The most important problems are:
- having all validations to be transferred correctly. It is not clear copy-paste unfortunately even though some settings in both libs match
- conditional validations are the problem as well: our current solution supports some "hacks" in order to make validation conditional. For instance have an acceptance validator with an `if` function. I tried to get rid of that and use a custom validator.
- I believe we've lost i18n support. So this part should be verified as well.

To sum up, maybe forking would be a good quick solution.

**Comment by donaldwasserman** [comment]
@stukalin - What would be the best approach to forking this? Should someone who have repository creation permissions fork it to the Hospital Run Repo? It's probably better than running it off someone's personal fork.

**Comment by stukalin** [comment]
That'd be the best option. I don't have sufficient right, I've just verified that. @jglovier, could you help forking the https://github.com/DockYard/ember-validations to our repo?

**Comment by stukalin** [comment]
So now it's only matter of applying the fix mentioned in the very first message to our fork and reference it from the main app.

**Comment by donaldwasserman** [comment]
I'm going to try to get that done this morning (well, morning East Coast US).
